OsmAnd-java/src/main/java/net/osmand/router/TransportRoutePlanner.java

```diff
@@ -785,18 +794,23 @@ public TransportRoutingContext(TransportRoutingConfiguration cfg, NativeLibrary
 			
 			// could be global ?
 			TLongObjectHashMap<TransportStop> loadedTransportStops = new TLongObjectHashMap<TransportStop>();
-			TIntObjectHashMap<TransportRoute> localFileRoutes = new TIntObjectHashMap<>();
+			TIntObjectHashMap<TransportRoute> localFileRoutes = new TIntObjectHashMap<>(); //reference, route
 			for (BinaryMapIndexReader r : routeMap.keySet()) {
 				sr.clearSearchResults();
 				List<TransportStop> stops = r.searchTransportIndex(sr);
 
 				localFileRoutes.clear();
 				mergeTransportStops(r, loadedTransportStops, stops, localFileRoutes, routeMap.get(r));
-
+					
 				for (TransportStop stop : stops) {
+					// skip missing stops
+					if (stop.isMissingStop()) {
+						continue;
+					}
 					long stopId = stop.getId();
 					TransportStop multifileStop = loadedTransportStops.get(stopId);
 					int[] rrs = stop.getReferencesToRoutes();
+					// TODO what is this?
```

Comment:
TODO why do we skip missing stop?

Reply:
TODO why duplicate same code?

Reply:
I thought that we need to skip missing stops so that they were not used during route calculation as valid start/finish/change stops. But I could be wrong with the exact place of such skip. Will look into this.